src/utils.rs

```diff
@@ -207,6 +206,7 @@ pub(crate) fn random_bn_in_z_star<R: RngCore + CryptoRng>(
         .ok_or(RetryFailed)
 }
 
+// Returns x: BigNumber as a k256::Scalar mod k256_order
```

Comment:
The function already performed a modulo operation (I just made this clearer), so submitting too big of a BN shouldn't be an issue and any failures of this function should be bugs.

Reply:
Nice, thanks for clarifying.